pkg/api/validate/dynamic/dynamic.go

```diff
@@ -61,6 +61,7 @@ type dynamic struct {
 	checkAccessSubjectInfoCred azcore.TokenCredential
 	env                        env.Interface
 	azEnv                      *azureclient.AROEnvironment
+	byoNSG                     bool
```

Comment:
i think `allowBYONSG` is better as `byoNSG` doesn't state clearly whether this aro has BYO NSG or this subscription has BYO NSG Feature Flag enabled.